test_model.py

```diff
@@ -0,0 +1,61 @@
+from argparse import ArgumentParser
+
+from config_utils import load_config
+from data_processing import load_stop_words, pickle_load_from_file, pickle_save_data_to_file, load_queries_and_output
+from boolean_model import process_query_boolean
+from vectorial_model import process_vectorial_query, get_stats_collection
+
+
+def get_boolean_results(queries, inv_index, stop_words):
+    results = []
+    for query in queries:
+        results.append(process_query_boolean(query, inv_index, stop_words))
+    return results
+
+
+def get_vectorial_results(queries, inv_index, stop_words, stats_collection, args):
+    results = []
+    for query in queries:
+        results.append(process_vectorial_query(query, inv_index, stop_words, stats_collection,
+                                               args.weight_document, args.weight_query))
+    return results
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    subparser = parser.add_subparsers(dest="model")
+    boolean = subparser.add_parser("boolean")
+    vectorial = subparser.add_parser("vectorial")
+
+    vectorial.add_argument("--weight-query", type=str, choices=("boolean", "frequency"))
+    vectorial.add_argument("--weight-document", type=str,
+                           choices=("boolean", "frequency", "tf_idf_normalize", "tf_idf_logarithmic",
+                                    "tf_idf_logarithmic_normalize"))
+    args = parser.parse_args()
+    config = load_config()
```

Comment:
Could you load all config paths in appropriate variables here, just after loading the config ?
It is a good practice to make your program raise an error before any computation if a path is missing in the config.
Otherwise, if the path is missing, you're doing the computation and then it crashes before being able to store the result.

For example:
```
stop_words_path = config.get('stop_words_path', 'stop_words_path')
...
```

Reply:
Done